### ISSUE TITLE
Redis Unix Socket can not connect with broker url

### ISSUE THREAD

**Issue by mdminhaz2003** [celery/celery#8146]
# Checklist

- [x] I have verified that the issue exists against the `main` branch of Celery.
- [ ] This has already been asked to the [discussions forum](https://github.com/celery/celery/discussions) first.
- [ ] I have read the relevant section in the
  [contribution guide](https://docs.celeryq.dev/en/main/contributing.html#other-bugs)
  on reporting bugs.
- [ ] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Bug+Report%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical bug reports.
- [ ] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22PR+Type%3A+Bugfix%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed fixes.
- [ ] I have checked the [commit log](https://github.com/celery/celery/commits/main)
  to find out if the bug was already fixed in the main branch.
- [ ] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

## Mandatory Debugging Information

- [ ] I have included the output of ``celery -A proj report`` in the issue.
    (if you are not able to do this, then at least specify the Celery
     version affected).
- [ ] I have verified that the issue exists against the `main` branch of Celery.
- [ ] I have included the contents of ``pip freeze`` in the issue.
- [ ] I have included all the versions of all the external dependencies required
  to reproduce this bug.

## Optional Debugging Information

- [ ] I have tried reproducing the issue on more than one Python version
  and/or implementation.
- [ ] I have tried reproducing the issue on more than one message broker and/or
  result backend.
- [ ] I have tried reproducing the issue on more than one version of the message
  broker and/or result backend.
- [ ] I have tried reproducing the issue on more than one operating system.
- [ ] I have tried reproducing the issue on more than one workers pool.
- [ ] I have tried reproducing the issue with autoscaling, retries,
  ETA/Countdown & rate limits disabled.
- [ ] I have tried reproducing the issue after downgrading
  and/or upgrading Celery and its dependencies.

#### Related Issues

- None

#### Possible Duplicates

- None

## Environment & Settings

**Celery version**:

<details>
<summary><b><code>celery report</code> Output:</b></summary>
<p>

```
```

</p>
</details>

# Steps to Reproduce

## Required Dependencies

- **Minimal Python Version**: 3.8
- **Minimal Celery Version**: 5.2.7
- **Minimal Kombu Version**: N/A or Unknown
- **Minimal Broker Version**: 4.5.2
- **Minimal Result Backend Version**: 4.5.2
- **Minimal OS and/or Kernel Version**: N/A or Unknown
- **Minimal Broker Client Version**: N/A or Unknown
- **Minimal Result Backend Client Version**: N/A or Unknown

### Python Packages

<details>
<summary><b><code>pip freeze</code> Output:</b></summary>
<p>

```
celery==5.2.7
flask_sqlalchemy==3.0.3
more_itertools==9.1.0
python-dotenv==1.0.0
redis==4.5.2
requests==2.28.2
SQLAlchemy==2.0.7
psycopg2-binary==2.9.5
supervisor==4.2.5
```

</p>
</details>

## Minimally Reproducible Test Case

<details>
<p>

```python
```

</p>
</details>

# Expected Behavior
 -------------- celery@moonwalk.hostns.io v5.2.7 (dawn-chorus)
--- ***** -----
-- ******* ---- Linux-4.18.0-372.19.1.lve.el8.x86_64-x86_64-with-glibc2.2.5 2023-03-22 13:44:54
- *** --- * ---
- ** ---------- [config]
- ** ---------- .> app:         core.create_app:0x7fc35bc9e790
- ** ---------- .> transport:   redis+socket:///home/pe3co/tmp/redis.sock
- ** ---------- .> results:     disabled://
- *** --- * --- .> concurrency: 16 (prefork)
-- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
--- ***** -----
 -------------- [queues]
                .> celery           exchange=celery(direct) key=celery

[tasks]
  . process_zipfile

# Actual Behavior

**Comment by woutdenolf** [comment]
redis-py 4.5.2 has a UDS bug (https://github.com/redis/redis-py/issues/2629). This should fix the issue: `pip install "redis>4.5.3"`

**Comment by auvipy** [comment]
thanks for the report and verification!
